### ISSUE TITLE
Send SMS message fails with topic not found

### ISSUE THREAD

**Issue by nykolaslima** [spulec/moto#3251]
SNS allows publish directly to a phone number in order to send SMS messages.

Moto SNS does not support it since it gives an exception of topic not found when using publish operation directly to a phone number.

e.g.
```
self.sns_client.publish(
                    PhoneNumber="+5511964448877",
                    Message="testing SMS message")
```

This code throws the following error:
```
botocore.errorfactory.InvalidParameterValueException: An error occurred (ParameterValueInvalid) when calling the Publish operation: Could not find topic associated with phone number
```

**Comment by nykolaslima** [comment]
As a workaround, you could create an SNS topic and subscribe to it using the phone number:
```
        sns = mock_sns()
        sns.start()

        sns_client = boto3.client('sns')
        sns_client.create_topic(Name="some-topic")
        resp = sns_client.create_topic(Name="some-topic")
        arn = resp["TopicArn"]
        sns_client.subscribe(TopicArn=arn, Protocol="sms", Endpoint='+5511964448877')
```

**Comment by kam193** [comment]
Looks like duplicate of #2198

**Comment by kam193** [comment]
@bblommers This can be also closed by #3253

**Comment by bblommers** [comment]
Thanks for raising this @nykolaslima - this is now fixed as of moto >= 1.3.15.dev1027

**Comment by nykolaslima** [comment]
Thank you ❤
